### PR TITLE
[NEW] Support Livechat iframe API

### LINKED ISSUE #55: Create support for iFrame API
We need to migrate support for iFrame API from our older version to new Livechat client.

### PR DESCRIPTION
**Pull request by renatobecker**
This PR provides a new feature on new client, which will support the iframe API methods.

Closes #55.

### CODE REVIEW

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging e8a700c851fed2882e3178f9fdcef35b98fb0410 into d85e1a651f3258e2f73d1828b4abb237b487e049 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-466236fd7c45c7ec5a62e36b4bcd647d527a468e)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 5 alerts** when merging 266246f0883dcc3c163790d95554642a2a61699b into 2f4db8ae4554b6b23927b883c5c73ce8aef8d92c - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-d2b3bd8e9562f1f586ec27a4f4565f9aa116a9d9)

**new alerts:**

* 5 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 5 alerts** when merging 947842667283b1ae074a605540a3962973f4d0e6 into 089804538a593d778ea0ce83574739bd704e717b - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-0e7b179b5a73d57fec193237824b1b4a4b32c85d)

**new alerts:**

* 5 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/components/App/index.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ import { Provider as StoreProvider, Consumer as StoreConsumer } from '../../stor
 import { loadConfig } from '../../lib/main';
 import CustomFields from '../../lib/customFields';
 import Triggers from '../../lib/triggers';
+import Hooks from '../../lib/hooks';
+import { parentCall } from '../../lib/parentCall';
+import * as RocketChat from '../../lib/rocketchat.js';
```
[the remark this answers is not in the document]
Comment (renatobecker) [reply src/components/App/index.js]: Sry, my bad.

**Comment by ggazzo** [comment]
This pull request **introduces 4 alerts** when merging 19b07b09c97b466366aee81273e1df01d60945c7 into 94f8be16eddfae8c0c40452cefaaee6f64209c51 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-374645e02693db7782d4722053dd818cc32924b0)

**new alerts:**

* 4 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by tassoevan** [comment]
I'm getting an error when running it using the current widget script:
captura de tela 2019-01-07 as 16 45 17

**Comment by renatobecker** [comment]
@tassoevan, can you please test it again?

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging 19514e789ad8e725c6222207c3c20bf2b3c9b09a into 06b93c0715e7d573d81cc82077a37cd52b4743b9 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-8d7df93c3513f7aba5773870380805e9d574a7f4)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging 19514e789ad8e725c6222207c3c20bf2b3c9b09a into 3ce3f8f7acaf4652730c6cd1d2c971a0c799647e - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-884732d377bb3b685158762308953053014e5a1f)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging 1756e760c86ef72366135e5338b7dbcaa0267c64 into 0d34ccc33f24c8dbc3057387176546369c672927 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-18eab74a61bd1508a7d9fd998ea781963bcba5ea)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/lib/hooks.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		SDK.sendVisitorNavigation({ token, rid: room._id, pageInfo: { change, title, location: { href } } });
+	},
+
+	setCustomField(key, value, overwrite = true) {
+		CustomFields.setCustomField(key, value, overwrite);
+	},
+
+	setTheme(newTheme = {}) {
+		const { iframe, iframe: { theme } } = store.state;
+		if (newTheme.color) {
```
Comment (ggazzo) [inline src/lib/hooks.js]: create the object first and after call just one `setState` to avoid multiple re renders

Comment (renatobecker) [reply src/lib/hooks.js]: @ggazzo, fixed!

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging d759b440ac7cd873ea147a28453e56a54f1ac698 into e3d1a18fb0ff96a84cc6c5272d1d4c50d6350524 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-4c2fd323073f47e69f0470233a00c059ba3da053)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging 32d51e05730861a8382d69997a31b3b4ca98d51e into f5b02cbe7b69a6fbdd8faf8b12e145a314af1ba3 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-1b6b53cffb47b2c17233ab9206e47a52335698ad)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging 3e926972a4ab4da3037d6993f8f445eb5b843762 into f5b02cbe7b69a6fbdd8faf8b12e145a314af1ba3 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-faf990940a1e0734ee77c99684c453f030016f28)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging b94c35cb5fc5e9c6752ec8849556786159e7d813 into 0b295814b7e7bfb166044ab16de0fc58eade0883 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-570f14e9c2e0ca5f6270bc27676478793ff802cd)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging 2c0e9dec16b262de29f3a68dbdce820a110a8278 into 0b295814b7e7bfb166044ab16de0fc58eade0883 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-7a4a7850f29cc88cab561f1a7c4fa638fee8ce8a)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 1 alert** when merging 8c078f017e3399ab0253d6ed47476bc1ef76467d into 0b295814b7e7bfb166044ab16de0fc58eade0883 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-38397dcf9399e5f78d96db5ef6f5e1c5b836e2b7)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ggazzo** [comment]
This pull request **introduces 2 alerts** when merging 063ba8f0cbd33666daf84e3aeda6bdcfb3384e0f into 0b295814b7e7bfb166044ab16de0fc58eade0883 - [view on LGTM.com](https://lgtm.com/projects/g/RocketChat/Rocket.Chat.Livechat/rev/pr-7d3cbb37c45e8956b87016b8a49cf58b9b58ad7d)

**new alerts:**

* 2 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**